### PR TITLE
GODRIVER-2631 - Always close internal.DefaultHTTPClient on Client.Disconnect

### PR DESCRIPTION
**Pull request by edaniels**
Also addresses GODRIVER-2623

### CODE REVIEW

**Comment by edaniels** [comment]
@matthewdale my bad on last PR, I missed the crucial point upon which to close the client 🙃 . Just verified this works in relation to the connections lingering from OCSP. Figured I'd knock another ticket out while I was at it.

File: internal/http.go [hunk 1]
```diff
@@ -8,19 +8,11 @@ package internal // import "go.mongodb.org/mongo-driver/internal"
...
 var DefaultHTTPClient = &http.Client{
-	// TODO(GODRIVER-2623): Use "http.DefaultTransport.Clone" once we change the minimum supported Go version to 1.13.
-	Transport: &http.Transport{
-		Proxy:                 http.ProxyFromEnvironment,
-		MaxIdleConns:          100,
-		IdleConnTimeout:       90 * time.Second,
-		TLSHandshakeTimeout:   10 * time.Second,
-		ExpectContinueTimeout: 1 * time.Second,
-	},
+	Transport: http.DefaultTransport.(*http.Transport).Clone(),
```
Comment (matthewdale) [inline internal/http.go:15]: We need to omit this change until after the v1.11.0 release. We're changing the documented minimum Go version in v1.11.0, but not including any actual breaking changes until the next release. Please remove this and add it in a separate PR which we can merge after the v1.11.0 release (scheduled for today).

File: mongo/client.go [hunk 2]
```diff
@@ -281,6 +281,10 @@ func (c *Client) Disconnect(ctx context.Context) error {
 		ctx = context.Background()
 	}
 
+	if c.httpClient == internal.DefaultHTTPClient {
+		defer internal.CloseIdleHTTPConnections(c.httpClient)
+	}
```
Comment (matthewdale) [inline mongo/client.go:286]: Nice fix!

**Review by matthewdale — COMMENTED** [review]
@edaniels thanks for the fix and the additional improvement!

I'd like to include the fix for always cleaning up the internal HTTP client in the v1.11.0 release (scheduled for today), but we can't include the `http.DefaultTransport.Clone` change until after that release. Please split these up into two PRs so we can merge them separately.

File: internal/http.go (hunk 1 above)
Comment (edaniels) [reply internal/http.go:15]: ah got it

**Comment by edaniels** [comment]
@matthewdale undid the transport change and made https://github.com/mongodb/mongo-go-driver/pull/1115

**Review by matthewdale — APPROVED** [review]
Awesome, thanks for splitting these PRs! Looks good, waiting for CI to finish 👍

**Pull request merged**